### ISSUE TITLE
Is this the correct formula for delta focal loss?

### ISSUE THREAD

**Issue by AlexeyAB** [marvis/pytorch-yolo2#79]
Hi @marvis 
Please, can you check, did I read your Focal Loss formulas correctly?

If for CE, delta is:
* `if (i == j)` then `delta = 1-p`
* `if (i != j)` then `delta = -p`

----

And if `focal_loss= -alpha * (pow( (1 - pt) , gamma)) * log(pt)`

Then for Focal Loss (when `gamm=2`), **delta** is:

* `if (i == j)` then `delta = (1-p)*` `alpha * (1 - pt) * (2 * pt * log(pt) + pt - 1)`
* `if (i != j)` then `delta = (-p)*` `alpha * (1 - pt) * (2 * pt * log(pt) + pt - 1)`

Where are:
* `pt = softmax(i)` - is a probability of the correct class id.
* `p = softmax(j)`
where is `i = label` truth class id.
